src/app/new-page/component/content-animation/content-animation.component.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  SimpleChanges
+} from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { PageService } from '../../service/page-service.service';
+import { Animation } from 'src/app/services/xml-parser-service/xmp-parser.service';
+
+@Component({
+  selector: 'app-content-new-animation',
+  templateUrl: './content-animation.component.html',
+  styleUrls: ['./content-animation.component.css']
+})
+export class ContentAnimationNewComponent implements OnChanges, OnDestroy {
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input() item: Animation;
+
+  private _unsubscribeAll = new Subject<any>();
+  animation: Animation;
+  ready: boolean;
+  anmResource: string;
+  anmViewItem: any;
+  hasEvents: boolean;
+  dir$: Observable<string>;
+  lottieOptions: any;
+
+  constructor(private pageService: PageService) {
+    this.dir$ = this.pageService.pageDir$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.animation = this.item;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  onAnimationClick(): void {
+    if (this.animation.events) {
+      let action = '';
+      this.animation.events.forEach((event, idx) => {
+        const value = event?.namespace
+          ? `${event.namespace}:${event.name}`
+          : event.name;
+        action += idx ? ` ${value}` : value;
+      });
+      this.pageService.formAction(action);
```

Comment:
we will definitely need to revisit how events are handled for knowgod.com. This will probably make the most sense to do when we are adding support for state events. I'll create a JIRA ticket for this

Reply:
see: GT-2034

Reply:
also see: GT-2035 regarding what clicking different content elements should do

Reply:
Thanks for making a ticket; we should definitely re-code this once we have state set up.